tensorboard/plugins/histogram/histograms_demo.py

```diff
@@ -107,18 +100,17 @@ def run_all(logdir, verbose=False, num_summaries=400):
         "two normal distributions.",
     )
 
-    summaries = tf.compat.v1.summary.merge_all()
 
-    # Setup a session and summary writer
-    sess = tf.compat.v1.Session()
-    writer = tf.summary.FileWriter(logdir)
-
-    # Setup a loop and write the summaries to disk
-    N = num_summaries
-    for step in range(N):
-        k_val = step / float(N)
-        summ = sess.run(summaries, feed_dict={k: k_val})
-        writer.add_summary(summ, global_step=step)
+def run_all(logdir, num_summaries=400):
+    """Generate a bunch of histogram data, and write it to logdir."""
+    tf.random.set_seed(0)
+    writer = tf.summary.create_file_writer(os.path.join(logdir, "my_run_name"))
+    with writer.as_default():
+        for step in range(num_summaries):
+            tf.summary.experimental.set_step(step)
```

Comment:
I think we are able to avoid using the "experimental" pieces of the API in this demo case.  Could we drop this experimental.set_step, and instead pass the `step` into the `run(k, step)`?  Then, each `tf.summary.histogram(..., step=step)` can specify the step without relying on the global `set_step`.

Reply:
+1 that it's a bit better to avoid `experimental` in a demo.  What Erik suggests here should work fine.

Reply:
Thanks for the review!